src/service/viewport-impl.js

```diff
@@ -1022,12 +1047,17 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop) {
+  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
     onDocumentReady(this.win.document, doc => {
       this.paddingTop_ = paddingTop;
       // Also tried `paddingTop` but it didn't work for `position:absolute`
       // on iOS.
       doc.body.style.borderTop = `${paddingTop}px solid transparent`;
+      if (opt_updateScrollPos) {
+        const adjScrollTop = this.getScrollTop() + 3 * paddingTop
+            - opt_lastPaddingTop);
+        this.setScrollTop(adjScrollTop);
```

Comment:
Also, please add `TODO` and explain that this formula only works when either `paddingTop` or `lastPaddingTop` is `0`.